### ISSUE TITLE
Deprecate LoopingCall.deferred

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8116]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8116|
|Type|enhancement|
|Created|2015-11-20 09:36:28Z|
|Branch|https://github.com/twisted/twisted/tree/loopincall-deferred-deprecate-8116|

and return it from start()

See [#7976](https://github.com/twisted/twisted/issues/7976)#comment:3

<details><summary>Searchable metadata</summary>

```
trac-id__8116 8116
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__branches_loopincall_deferred_deprecate_8116 branches/loopincall-deferred-deprecate-8116
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1448012188525436 1448012188525436
changetime__1451139248314533 1451139248314533
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

Please check the changes.

While looking at this I have also update the LoopingCall example to show how the deferred returned by start can be use.

I don't know if I should also define the setter... I hope not :)

I have no idea why the self.deferred , now self._deferred is set to None after being called.

I don't know how to use the @deferred decorator with a @property decorator. The property will generate a function and not a method.

Rather than raising a custom warning, I prefer to use a pre-defined method so that we have standard text message and add a bit more of meta-data to this call.

Please review and suggest changes.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Hi Adi,

@property and @deprecated don't seem to work together -- it makes an incorrect FQDN and the tests fail. The public ```getDeprecationWarningString``` also doesn't work for properties, as the FQDN lookup makes it refer it, and then causes an infinite loop, so we need to use the private one that takes a string.

I've spun the builders, lets see how this goes.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

As I changed it, putting this up for review, and someone other than Adi will have to review it :)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set owner to @adiroiban|
|-|-|

Looks great. Thanks for the fixes. I don't really have much feedback, since it looks like automated tests would have complained about anything I was looking for :).

Thanks so much adi for adding documentation of how to use this as well as fixing it!

Please land.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

Thanks! Will merge it.

There is [#8124](https://github.com/twisted/twisted/issues/8124) which tries to improve @deprecate decorator for properties and classes... that ticket is also blocking the documentation update [#8082](https://github.com/twisted/twisted/issues/8082)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban set status to `closed`|
|-|-|

(In [46566]) Merge loopincall-deferred-deprecate-8116: Deprecate LoopingCall.deferred.

Author: adiroiban, hawkowl
Reviewer: hawkowl, glyph
Fixes: [#8116](https://github.com/twisted/twisted/issues/8116)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban set status to `reopened`|
|-|-|

(In [46573]) Revert r46566: Deprecate LoopingCall.deferred.

The docstring was not complete and the deprecation tests were not complete.

Reopens: [#8116](https://github.com/twisted/twisted/issues/8116)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban commented|
|-|-|

(In [46574]) Branching to loopincall-deferred-deprecate-8116-2.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_10"></a>|@adiroiban commented|
|-|-|

It was my fault for merging without pulling Amber's changes. My bad.

I will merge again with the Amber changes, as those changes were already approved.

Sorry for the trouble.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_11"></a>|@adiroiban set status to `closed`|
|-|-|

(In [46575]) Merge loopincall-deferred-deprecate-8116: Deprecate LoopingCall.deferred.

Author: adiroiban, hawkowl
Reviewer: hawkowl, glyph
Fixes: [#8116](https://github.com/twisted/twisted/issues/8116)

It re-merge the branch at the latest revision.
In r46566 it was not merged on the latest branch revision.
